### PR TITLE
Add the application Review page

### PR DESCRIPTION
**Pull request by alexdesi**
### Context
We need to add a new page to review the Application

This PR adds:

- the new Review application page, only the Title for now.
- route
- controller

### Guidance to review
have a look to see if the routes and page name is a good choice.

### Link to Trello card

https://trello.com/c/F8GxGHB1/123-allow-users-to-submit-the-application-form

### CODE REVIEW

**Comment by alexdesi** [comment]
Screenshot 2019-10-14 at 17 07 48

**Comment by alexdesi** [comment]
Screenshot 2019-10-14 at 17 07 48

**Comment by alexdesi** [comment]
Screenshot 2019-10-14 at 17 07 23

File: app/views/candidate_interface/application_form/show.html.erb [hunk 1]
```diff
@@ -76,5 +76,8 @@
         <%= govuk_link_to 'TODO: Check your answers before submitting', '#', class: 'app-task-list__task-name' %>
       </li>
     </ol>
+
+    <%= link_to 'Check your answers before submitting', candidate_interface_application_review_path %>
```
Comment (tvararu) [inline app/views/candidate_interface/application_form/show.html.erb]: I think there's already a link for it, above on line 76 👀

File: config/routes.rb [hunk 2]
```diff
@@ -22,6 +22,7 @@
 
     scope :application do
       get '/' => 'application_form#show', as: :application_form
+      get '/application/review' => 'application_review#show'
```
Comment (chubberlisk) [inline config/routes.rb]: > URL for review:
> 
> Screenshot 2019-10-14 at 17 07 48

Are we sure we want this as the URL with the repetition of `/application`?

File: spec/system/candidate_interface/candidate_reviewing_application_spec.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+RSpec.feature 'Candidate reviews the answers' do
+  scenario 'Logged in candidate with no personal details' do
+    given_i_am_signed_in
+    and_i_visit_the_application_form_page
+    when_i_click_on_check_your_answers
+    then_i_can_review_my_application
+  end
+
+  def when_i_click_on_check_your_answers
+    click_link 'Check your answers before submitting'
```
Comment (chubberlisk) [inline spec/system/candidate_interface/candidate_reviewing_application_spec.rb]: Could this be a translation string?

File: spec/system/candidate_interface/candidate_reviewing_application_spec.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    visit candidate_interface_application_form_path
+  end
+
+  def given_i_am_signed_in
+    candidate = FactoryBot.create(:candidate)
+    login_as(candidate)
+  end
+
+  def then_i_can_review_my_application
+    expect(page).to have_content 'Review your application'
```
Comment (chubberlisk) [inline spec/system/candidate_interface/candidate_reviewing_application_spec.rb]: Could you use `t('page_titles.review_application')` here?

File: config/routes.rb (hunk 2 above)
Comment (alexdesi) [reply config/routes.rb]: of course not ;) I'll fix it, thanks!

File: spec/system/candidate_interface/candidate_reviewing_application_spec.rb [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  def when_i_click_on_check_your_answers
+    click_link 'Check your answers before submitting'
+  end
+
+  def and_i_visit_the_application_form_page
+    visit candidate_interface_application_form_path
+  end
+
+  def given_i_am_signed_in
+    candidate = FactoryBot.create(:candidate)
```
Comment (duncanjbrown) [inline spec/system/candidate_interface/candidate_reviewing_application_spec.rb]: `config.include FactoryBot::Syntax::Methods` in `rails_helper.rb` makes `FactoryBot.` redundant here

File: spec/system/candidate_interface/candidate_reviewing_application_spec.rb (hunk 3 above)
Comment (alexdesi) [reply spec/system/candidate_interface/candidate_reviewing_application_spec.rb]: yes, I'm happy to add it to next prs.

File: spec/system/candidate_interface/candidate_reviewing_application_spec.rb (hunk 5 above)
Comment (alexdesi) [reply spec/system/candidate_interface/candidate_reviewing_application_spec.rb]: I removed it from (almost) over the other places

File: spec/system/candidate_interface/candidate_reviewing_application_spec.rb (hunk 4 above)
Comment (alexdesi) [reply spec/system/candidate_interface/candidate_reviewing_application_spec.rb]: as discussed, we'll to it later.

**Pull request merged**